Sources/FoundationEssentials/FileManager/FileManager+Files.swift

```diff
@@ -576,20 +714,51 @@ extension _FileManagerImpl {
                 .systemNumber : _writeFileAttributePrimitive(fsNumber, as: UInt.self)
             ]
         }
+#endif
     }
     
     func setAttributes(
         _ attributes: [FileAttributeKey : Any],
         ofItemAtPath path: String
     ) throws {
+        let mode = _readFileAttributePrimitive(attributes[.posixPermissions], as: UInt.self)
+        let immutable = _readFileAttributePrimitive(attributes[.immutable], as: Bool.self)
+        let appendOnly = _readFileAttributePrimitive(attributes[.appendOnly], as: Bool.self)
+
+#if os(Windows)
+        try path.withNTPathRepresentation {
+            if immutable != nil || appendOnly != nil {
+                // Setting these flags is not supported on this platform
+                throw CocoaError.errorWithFilePath(.featureUnsupported, path)
```

Comment:
If these attributes aren't supported on windows, I think we should wrap their declarations in `#if !os(Windows)` so that it's clear to clients at compile time

Reply:
TBH, I don't know about these. I'm just replicating the non-Darwin path.